tests/integration/test_integration.py

```diff
@@ -1,28 +1,140 @@
 #!/usr/bin/env python3
-# Copyright 2021 Canonical Ltd.
+# Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-"""Integration tests for the orc8r-controller rock.
+"""Integration tests for the lte-controller rock.
 
-These tests assume that the orc8r-controller container is running and available at the url:port
+These tests assume that the lte-controller container is running and available at the url:port
 defined below.
 """
 
+import glob
+import logging
+import subprocess
+import unittest
+from time import sleep, time
+
+import docker  # type: ignore[import]
 import requests  # type: ignore[import]
+import yaml
 
-import unittest
+ROCKCRAFT_FILE_PATH = "../rockcraft.yaml"
 
-ORC8R_CONTROLLER_DOCKER_URL = 'http://localhost'
-ORC8R_CONTROLLER_DOCKER_PORT = 8080
+LTE_CONTROLLER_DOCKER_URL = "http://localhost"
+LTE_CONTROLLER_DOCKER_PORT = 8080
 
+POSTGRES_USER = "username"
+POSTGRES_PASSWORD = "password"
+POSTGRES_DB = "magma"
 
-class TestOrc8rControllerRock(unittest.TestCase):
-    """Integration tests for the orc8r-controller rock."""
+logger = logging.getLogger(__name__)
 
-    def test_given_orc8r_controller_container_is_running_when_http_get_then_hello_message_is_returned(  # noqa: E501
-        self
-    ):
-        response = requests.get(f"{ORC8R_CONTROLLER_DOCKER_URL}:{ORC8R_CONTROLLER_DOCKER_PORT}")
 
-        assert response.status_code == 200
-        assert "hello" in response.text
+class TestLTEControllerRock(unittest.TestCase):
+    """Integration tests for the lte-controller rock."""
+
+    def setUp(self):
+        """Runs containers under test."""
+        self._move_rock_to_docker_registry()
+
+        self.client = docker.from_env()
+        self.network = self.client.networks.create(
+            "bridge_network",
+            driver="bridge",
+        )
+
+        self._run_postgres_container()
+
+        timeout = 15
+        start_time = time()
+        while (time() - start_time) < timeout:
+            if not self._postgres_ready():
+                sleep(1)
+            else:
+                self._run_orc8r_lte_controller_container()
+                return
+        raise TimeoutError("Postgres container is not ready after 15 seconds.")
+
+    @staticmethod
+    def _get_image_name_and_version() -> tuple:
+        """Fetches image name and version from rockcraft.yaml."""
+        with open(ROCKCRAFT_FILE_PATH, "r") as file:
+            data = yaml.safe_load(file)
+            image_name = data["name"]
+            version = data["version"]
+        return (image_name, version)
+
+    @property
+    def _host_ip(self) -> str:
```

Comment:
I don't think it makes sense for `_host_ip` to be a property of the `TestLTEControllerRock` class. To make it simple I'd simply use a method called `_get_container_ip(name: str) -> str`.

Reply:
Done.